src/libraries/System.Text.Json/src/System/Text/Json/Serialization/JsonSerializerOptions.Caching.cs

```diff
@@ -172,7 +175,12 @@ public CachingContext(JsonSerializerOptions options, int hashCode)
             // If changing please ensure that src/ILLink.Descriptors.LibraryBuild.xml is up-to-date.
             public int Count => _jsonTypeInfoCache.Count;
 
-            public JsonTypeInfo? GetOrAddJsonTypeInfo(Type type) => _jsonTypeInfoCache.GetOrAdd(type, _jsonTypeInfoFactory);
+            public JsonTypeInfo? GetOrAddJsonTypeInfo(Type type) =>
+#if NETCOREAPP
+                _jsonTypeInfoCache.GetOrAdd(type, static (type, options) => options.GetTypeInfoNoCaching(type), Options);
```

Comment:
I thought you thought this was overkill, hence why I removed it from my PR.  Did I misunderstand, or you changed your mind?

Reply:
Changed my mind :-)